### PR TITLE
[ML-27304][WIP] Remove unused env params in Feature Store ETL and Training notebooks

### PR DESCRIPTION
**Pull request by zhe-db**
Removed unused env params in Feature Store ETL and Training notebooks

- {{cookiecutter.project_name}}/notebooks/GenerateAndWriteFeatures.py
- {{cookiecutter.project_name}}/notebooks/TrainWithFeatureStore.py
- And upstream TF files and CICD Yml files

The reason we remove the env param is because

- Staging and production models are already differentiated by their name in the TF config files: https://github.com/databricks/mlops-stack/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/databricks-config/prod/training-job.tf#L22
- There is no need for differentiating feature tables by env and there is no use of env in downsteam APIs that we called.

Additional manual tests:
- Ran the notebooks on Databricks.
- Setup a dummy project on GitHub and created a dummy PR that triggers CI.

Testing with MLOps Azure CUJ Repo: https://github.com/databricks/mlops-azure-cuj/pull/55

### CODE REVIEW

**Review by arpitjasa-db — COMMENTED** [review]
Excellent work on this @zhe-db! Could we add in the description a bit more about why we're making this change instead of plumbing the params through for example? And it would be good to do a bit of testing to confirm the notebooks can still run as expected

**Comment by arpitjasa-db** [comment]
Also don't forget to change [write-feature-table-job.tf](https://github.com/databricks/mlops-stack/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/databricks-config/prod/write-feature-table-job.tf) and [training-job.tf](https://github.com/databricks/mlops-stack/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/databricks-config/prod/training-job.tf) in both `staging` and `prod` to not pass in the `env` variables

**Comment by zhe-db** [comment]
> Also don't forget to change [write-feature-table-job.tf](https://github.com/databricks/mlops-stack/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/databricks-config/prod/write-feature-table-job.tf) and [training-job.tf](https://github.com/databricks/mlops-stack/blob/main/%7B%7Bcookiecutter.project_name%7D%7D/databricks-config/prod/training-job.tf) in both `staging` and `prod` to not pass in the `env` variables

Done, also remove the occurrences in other tf and yaml files.

File: {{cookiecutter.project_name}}/databricks-config/staging/training-job.tf
Comment (mingyu89) [inline {{cookiecutter.project_name}}/databricks-config/staging/training-job.tf:28]: For notebooke Train, the env is needed.

File: {{cookiecutter.project_name}}/databricks-config/prod/training-job.tf
Comment (mingyu89) [inline {{cookiecutter.project_name}}/databricks-config/prod/training-job.tf:28]: For notebooke Train, the env is needed.

**Review by mingyu89 — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_name}}/.azure/devops-pipelines/tests-ci.yml
Comment (mingyu89) [inline {{cookiecutter.project_name}}/.azure/devops-pipelines/tests-ci.yml:98]: Ditto

File: {{cookiecutter.project_name}}/databricks-config/staging/training-job.tf
Comment (zhe-db) [reply {{cookiecutter.project_name}}/databricks-config/staging/training-job.tf]: thanks for the catch Mingyu, fixed.

**Review by mingyu89 — COMMENTED** [review]
LGTM. I'll leave it to @arpitjasa-db to take another look and approve.

File: {{cookiecutter.project_name}}/.github/workflows/run-tests.yml
Comment (arpitjasa-db) [inline {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: Wait why are we enabling `test_mode` by default for feature store?

Comment (zhe-db) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: yeah, `test_mode` is not needed, the payload for feature store ones should be empty

Comment (arpitjasa-db) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: Wait why do we need this at all? Can we revert?

Comment (arpitjasa-db) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: It looks weird to just have notebook params be empty. Can we rewrite this so the params are only added if it doesn't include feature store?

Comment (zhe-db) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: Because if [cookiecutter.include_feature_store is true, we set the local-notebook-path to be `otebooks/TrainWithFeatureStore.py`](https://github.com/databricks/mlops-stack/blob/3f9251ce8c22379852405f5d55dc6716dd7c71a8/%7B%7Bcookiecutter.project_name%7D%7D/.github/workflows/run-tests.yml#L52), which does not take `env` nor `test_mode`?

Comment (zhe-db) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: cool

> It looks weird to just have notebook params be empty. Can we rewrite this so the params are only added if it doesn't include feature store?

cool

File: {{cookiecutter.project_name}}/databricks-config/staging/training-job.tf
Comment (vladimirk-db) [inline {{cookiecutter.project_name}}/databricks-config/staging/training-job.tf:21]: Nitpick and note for future consideration: the environment value is used implicitly in this notebook. It's great that we are cleaning up the explicit unused param env, we should audit and figure out how to make "env" naming consistent across these notebooks in the future. For instance, models have environment prefixed in their name, but feature store tables don't.

**Review by vladimirk-db — APPROVED** [review]
Nicely done, @zhe-db !  LGTM

File: {{cookiecutter.project_name}}/.github/workflows/run-tests.yml
Comment (arpitjasa-db) [inline {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: Nit: But what if we make this `== "no"` instead? Bit cleaner code and easier for our customers to understand

**Review by mingyu89 — APPROVED** [review]

**Review by arpitjasa-db — APPROVED** [review]
Looks great thanks for all the quick fixes @zhe-db! Let's confirm it all works though before merging

File: {{cookiecutter.project_name}}/.github/workflows/run-tests.yml
Comment (mingyu89) [inline {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: @arpitjasa-db It's fine as long as the condition is consistent across all places. Since other places are using `cookiecutter.include_feature_store == "yes"`, We should use `cookiecutter.include_feature_store != "yes"` here.

**Review by mingyu89 — CHANGES_REQUESTED** [review]

File: {{cookiecutter.project_name}}/.github/workflows/run-tests.yml
Comment (arpitjasa-db) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: Wait @mingyu89 I disagree. In other places in stacks like [here](https://src.dev.databricks.com/databricks/mlops-stack/-/blob/%7B%7Bcookiecutter.project_name%7D%7D/docs/mlops-setup.md?L46:21), we use `"no"`. I think it would be better to stay consistent with this convention.

Comment (mingyu89) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: @arpitjasa-db  Ok. Then I'm Ok with fixing it in another PR. When we only check against one string, we are sure the condition is consistent for all places.

**Review by mingyu89 — APPROVED** [review]

File: {{cookiecutter.project_name}}/.github/workflows/run-tests.yml
Comment (mingyu89) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: It's not likely, but for example, if the value is "Yes" or "true", we treat it as "no feature store" across all places.(easier for user to figure out the outcome and the reason)

Comment (arpitjasa-db) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: Yeah I see your point, maybe we should do a better job instead of enforcing the schema described [here](https://src.dev.databricks.com/databricks/mlops-stack/-/blob/cookiecutter.json?L11:4).

Comment (mingyu89) [reply {{cookiecutter.project_name}}/.github/workflows/run-tests.yml:80]: @arpitjasa-db  We can use this once 2.2.0 is available. https://cookiecutter.readthedocs.io/en/latest/advanced/boolean_variables.html

**Pull request merged**
